src/lib.rs

```diff
@@ -71,6 +71,13 @@
 //! Guessing the proportion of ASCII vs. non-ASCII should be particularly
 //! feasible.
 
+#![no_std]
+
+#[cfg(feature = "alloc")]
+extern crate alloc;
+#[cfg(feature = "std")]
+extern crate std;
```

Comment:
Why is this useful instead of always depending on `alloc`?

Reply:
emmm, if you don't want to support `serde/std`, there seems not to be any difference. 
But if `no_std` then `extern crate alloc` by default and remove `alloc` feature, will it be better?

Reply:
Will always depending on `alloc` be better? I cannot decide.